src/guardian/guardian.service.ts

```diff
@@ -124,110 +134,249 @@ export class GuardianService implements OnModuleInit {
     this.logger.log('New staking router state cycle start');
 
     try {
+      // Fetch the minimum required data to make an early exit
+      // fetch data from Keys api
       const { data: operatorsByModules, meta } =
-        await this.stakingRouterService.getOperatorsAndModules();
+        await this.keysApiService.getOperatorListWithModule();
 
       const {
         elBlockSnapshot: { blockHash, blockNumber },
       } = meta;
 
+      // contracts init
       await this.repositoryService.initCachedContracts({ blockHash });
 
-      if (
-        !this.blockGuardService.isNeedToProcessNewState({
-          blockHash,
-          blockNumber,
-        })
-      ) {
-        this.logger.debug?.(
-          `The block has not changed since the last cycle. Exit`,
-          {
-            blockHash,
-            blockNumber,
-          },
-        );
-        return;
-      }
+      const isNewBlock = this.blockGuardService.isNeedToProcessNewState({
+        blockHash,
+        blockNumber,
+      });
+
+      if (!isNewBlock) return;
 
       const stakingModulesCount = operatorsByModules.length;
 
       this.logger.log('Staking modules loaded', {
         modulesCount: stakingModulesCount,
       });
 
-      await this.depositService.handleNewBlock(blockNumber);
+      // fetch all lido keys
+      const { data: lidoKeys, meta: currMeta } =
+        await this.keysApiService.getKeys();
 
-      // TODO: e2e test 'node operator deposit frontrun' shows that it is possible to find event and not save in cache
-      const blockData = await this.blockGuardService.getCurrentBlockData({
-        blockHash,
-        blockNumber,
-      });
+      // check that there were no updates in Keys Api between two requests
+      this.keysApiService.verifyMetaDataConsistency(
+        meta.elBlockSnapshot.lastChangedBlockHash,
+        currMeta.elBlockSnapshot.lastChangedBlockHash,
+      );
```

Comment:
The `/v2/modules/operators/keys` endpoint returns data in a different format compared to other endpoints. It will require us to make significant changes. Additionally, it is quite experimental. 
However, I agree that the contract cache should be initialized after verifying the consistency of requests.

Reply:
part with cache init fixed here fix/val-1200-keys-events-fetch-impr 
part with fetching data from keys api here https://github.com/lidofinance/lido-council-daemon/pull/238